tests/playwright/src/model/pages/create-machine-page.ts

```diff
@@ -0,0 +1,134 @@
+  readonly userModeNetworkingCheckbox: Locator;
+  readonly startNowCheckbox: Locator;
+  readonly closeButton: Locator;
+  readonly createMachineButton: Locator;
+
+  constructor(page: Page) {
+    super(page);
+    this.heading = this.page.getByRole('heading', { name: 'Create Podman Machine' });
+    this.podmanMachineConfiguration = this.page.getByRole('form', { name: 'Properties Information' });
+    this.podmanMachineName = this.podmanMachineConfiguration.getByRole('textbox', { name: 'Name' });
+    this.imagePathBox = this.podmanMachineConfiguration.getByRole('textbox', { name: 'Image Path (Optional) ' });
+    this.browseImagesButton = this.podmanMachineConfiguration.getByRole('button', {
+      name: 'button-Image Path (Optional)',
+    });
+    this.podmanMachineCPUs = this.podmanMachineConfiguration.getByRole('slider', { name: 'CPU(s)' });
+    this.podmanMachineMemory = this.podmanMachineConfiguration.getByRole('slider', { name: 'Memory' });
+    this.podmanMachineDiskSize = this.podmanMachineConfiguration.getByRole('slider', { name: 'Disk size' });
+    this.rootPriviledgesCheckbox = this.podmanMachineConfiguration.getByRole('checkbox', {
+      name: 'Machine with root privileges',
+    });
+    this.userModeNetworkingCheckbox = this.podmanMachineConfiguration.getByRole('checkbox', {
+      name: 'User mode networking',
+    });
+    this.startNowCheckbox = this.podmanMachineConfiguration.getByRole('checkbox', { name: 'Start the machine now' });
+    this.closeButton = this.page.getByRole('button', { name: 'Close' });
+    this.createMachineButton = this.page.getByRole('button', { name: 'Create' });
+  }
+
+  async createMachine(
+    machineName: string,
+    isRootful: boolean = true,
+    enableUserNet: boolean = false,
+    startNow: boolean = true,
+    setAsDefault: boolean = true,
+  ): Promise<ResourcesPage> {
+    await playExpect(this.podmanMachineConfiguration).toBeVisible();
+    await this.podmanMachineName.fill(machineName);
+
+    if (isRootful !== (await this.isEnabled(this.rootPriviledgesCheckbox))) {
+      await this.switchCheckbox(this.rootPriviledgesCheckbox);
+    }
+
+    if (enableUserNet !== (await this.isEnabled(this.userModeNetworkingCheckbox))) {
+      await this.switchCheckbox(this.userModeNetworkingCheckbox);
+    }
+
+    if (startNow !== (await this.isEnabled(this.startNowCheckbox))) {
+      await this.switchCheckbox(this.startNowCheckbox);
+    }
+
+    await this.createMachineButton.click();
+
+    const successfulCreationMessage = this.page.getByText('Successful operation');
+    const goBackToResourcesButton = this.page.getByRole('button', { name: 'Go back to resources' });
+
+    await playExpect(successfulCreationMessage).toBeVisible({ timeout: 200_000 });
+    await playExpect(goBackToResourcesButton).toBeVisible();
+    await this.page.waitForTimeout(2_000); //wait for dialog to appear, in the needed cases
+    await this.handleConnectionDialog(setAsDefault);
```

Comment:
This is not the best approach, you should be waiting until the element that you need is available, I would say that in this case you should use `await playExpect(setAsDefault).toBeVisible()` instead of waitForTimeout(), which could very well introduce flakyness.

By waiting on the specific element it also allows the implementation of handleConnectionDialog() method to be refactored, as pointed out below.